release.js

```diff
@@ -36,15 +36,37 @@ module.exports = async function ({ github, context, inputs }) {
 
   const { opticUrl, npmTag, version, id } = releaseMeta
 
+  const run = runSpawn()
   if (!pr.merged) {
-    return github.rest.repos.deleteRelease({
-      owner,
-      repo,
-      release_id: id,
-    })
+    const branchName = `release/${version}`
+    const [branchPromise, releasePromise] = await Promise.allSettled([
+      run('git', ['push', 'origin', '--delete', branchName]),
+      github.rest.repos.deleteRelease({
+        owner,
+        repo,
+        release_id: id,
+      }),
+    ])
+
+    if (
+      branchPromise.status === 'rejected' &&
+      releasePromise.status === 'rejected'
+    ) {
+      logError(branchPromise.reason.message)
+      logError(releasePromise.reason.message)
+      core.setFailed(`Could not delete the branch or the release.`)
+    } else if (branchPromise.status === 'rejected') {
+      logError(branchPromise.reason.message)
+      core.setFailed(`Could not delete branch ${branchName}`)
+    } else if (releasePromise.status === 'rejected') {
+      logError(releasePromise.reason.message)
+      core.setFailed(`Could not delete release ${id}`)
+    }
```

Comment:
let's simplify this logic please:

- extract error messages, if any
- put them in an array
- if the array has elements, set the action failed with the joint error messages

Reply:
I have updated the logic and while it is simpler, I feel it gives the user a bit less info. Let me know what you think.